### PR TITLE
Oidc primer part 1

### PR DESCRIPTION
**Pull request by dogeared**
## Description:
- Added dogeared avatar/author info  and oidc primer part 1 post

### CODE REVIEW

File: _source/_posts/2017-07-25-oidc-primer-part-1.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
+---
+layout: blog_post
+title: 'Identity, Claims, & Tokens – An OpenID Connect Primer, Part 1 of 3'
+author: dogeared
+tags: [oauth, oauth2, OpenID, OpenID Connect, oidc]
+---
+
+In the beginning, there were proprietary approaches to working with external identity providers for authentication and authorization. Then came SAML (Security Assertion Markup Language) – an open standard using XML as its message exchange type. Then, there was OAuth and OAuth 2.0 (henceforth OAuth2) – also open as well as being a modern, RESTful approach to authorization using JSON as its medium. And now, the holy grail of “secure delegated access” OpenID Connect, which runs on top of OAuth2.
```
Comment (mraible) [inline _source/_posts/2017-07-25-oidc-primer-part-1.md]: In my blog post on OAuth, I used "OAuth" instead of "OAuth2". Should we be consistent?

Comment (jmelberg-okta) [inline _source/_posts/2017-07-25-oidc-primer-part-1.md]: In the following sections, we refer to only the acronym for OpenID Connect. Can we append the acronym (OIDC) when we mention OpenID Connect here so it is clear to the reader?

File: _source/_posts/2017-07-25-oidc-primer-part-1.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+
+Without secure, external authentication and authorization, you’d have to trust that every application, and every developer not only had your best interests and privacy in mind, but also *knew how to protect your identity* and was willing to keep up with security best practices. That’s a pretty tall order, right? With OIDC, you can use a trusted external provider to prove to a given application that you are who you say you are, without ever having to grant that application access to your credentials.
+
+OAuth2 leaves a lot of details up to implementers. For instance, it supports scopes, but scope names are not specified. It supports access tokens, but the format of those tokens are not specified. With OIDC, a number of specific scope names are defined that each produce different results. OIDC has both access tokens and ID tokens. An ID token must be JSON web token (JWT). Since the specification dictates the token format, it makes it easier to work with tokens across implementations.
```
Comment (jmelberg-okta) [inline _source/_posts/2017-07-25-oidc-primer-part-1.md]: I wish we could make this section 100pt font w/ marquee animations to signify its importance. We cannot stress this enough. Thanks for adding this in!

File: _source/_posts/2017-07-25-oidc-primer-part-1.md [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+In this blog series, I share a primer on OpenID Connect (OIDC). In the first post, we’ll review some key concepts around OIDC and tokens, explained in human terms. Then, we’ll look at OIDC in action with some specific code examples to highlight its value in the authentication and authorization ecosystem. Finally, we’ll dig into the guts of the different token types and how to control what goes into them.
```
Comment (jmelberg-okta) [inline _source/_posts/2017-07-25-oidc-primer-part-1.md]: As noted above, lets call out the acronym in the first paragraph.

File: _source/_posts/2017-07-25-oidc-primer-part-1.md [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+
+## All About Tokens
+
+With the foundation of scopes, claims, and response types, we can now talk about tokens! There are three types of tokens in OIDC: `id_token`, `access_token` and `refresh_token`.
+
+### ID Tokens
+
+An `id_token` is a [JWT](https://tools.ietf.org/html/rfc7519), per the [OIDC Specification](http://openid.net/specs/openid-connect-core-1_0.html#TokenResponse). This means that (a) identity information about the user is encoded right into the token and (b) the token can be definitively verified to prove that it hasn’t been tampered with. There’s a set of [rules](http://openid.net/specs/openid-connect-core-1_0.html#IDTokenValidation) in the specification for validating an `id_token`. Among the claims encoded in the `id_token` is an expiration (`exp`), which must be honored as part of the validation process. Additionally, the signature section of JWT is used in concert with a key to validate that the entire JWT has not been tampered with in any way.
```
Comment (jmelberg-okta) [inline _source/_posts/2017-07-25-oidc-primer-part-1.md]: Suggestion: Can we split this section into bullets?
---
An `id_token` is a [JWT](https://tools.ietf.org/html/rfc7519), per the [OIDC Specification](http://openid.net/specs/openid-connect-core-1_0.html#TokenResponse). This means that:
  - Identity information about the user is encoded right into the token.
  - The token can be definitively verified to prove that it hasn’t been tampered with.

There’s a set of [rules](http://openid.net/specs/openid-connect-core-1_0.html#IDTokenValidation) in the specification for validating an `id_token`. ...

**Comment by jmelberg-okta** [comment]
Re: OAuth vs OAuth2 -> I think we should use OAuth2 for the same reason we use OpenID Connect instead of OpenID.

Can we update this?

**Comment by dogeared** [comment]
Changed OAuth back to OAuth2

**Pull request merged**